lib/Proxy/AbstractProxy.php

```diff
@@ -42,46 +44,56 @@ abstract class AbstractProxy
      * initializes and runs the proxy class
      *
      * @access public
+     * @param Configuration $conf
      * @param string $link
      */
-    public function __construct(Configuration $conf, $link)
+    public function __construct(Configuration $conf, string $link)
     {
         if (!str_starts_with($link, $conf->getKey('basepath') . '?')) {
             $this->_error = 'Trying to shorten a URL that isn\'t pointing at our instance.';
             return;
         }
 
-        $data = $this->_getcontents($conf, $link);
+        $proxyUrl = $this->_getProxyUrl($conf);
 
-        if ($data == null) {
-            $this->_error = 'Error calling proxy. Probably a configuration issue';
-            error_log('Error calling proxy: ' . $this->_error);
+        if (empty($proxyUrl)) {
+            $this->_error = 'Error calling proxy. Probably a configuration issue, like wrong or missing config keys.';
+            error_log($this->_error);
             return;
         }
 
+        $data = file_get_contents($proxyUrl, false,
+            stream_context_create(
+                array(
+                    'http' => $this->_getProxyPayload($conf, $link),
+                )
+            )
+        );
+
         if ($data === false) {
             $http_response_header = $http_response_header ?? array();
             $statusCode           = '';
             if (!empty($http_response_header) && preg_match('/HTTP\/\d+\.\d+\s+(\d+)/', $http_response_header[0], $matches)) {
                 $statusCode = $matches[1];
             }
-            $this->_error = 'Error calling proxy. HTTP request failed. Status code: ' . $statusCode;
-            error_log('Error calling proxy: ' . $this->_error);
+            $this->_error = 'Error calling proxy. Probably a configuration issue, like wrong or missing config keys.';
+            error_log($this->_error . ' Status code: ' . $statusCode);
             return;
         }
 
         try {
-            $data = Json::decode($data);
+            $jsonData = Json::decode($data);
         } catch (Exception $e) {
             $this->_error = 'Error calling proxy. Probably a configuration issue, like wrong or missing config keys.';
             error_log('Error calling proxy: ' . $e->getMessage());
             return;
         }
 
-        $url = $this->_extractShortUrl($data);
+        $url = $this->_extractShortUrl($jsonData);
 
-        if ($url === null) {
-            $this->_error = 'Error calling proxy. Probably a configuration issue, like wrong or missing config keys.';
+        if ($url === null || empty($url)) {
+            $this->_error = 'Error parsing proxy response.';
```

Comment:
Hmm I guess the addition of:
> Probably a configuration issue, like wrong or missing config keys

Had the good intention to lead server/sysadmins to the correct point, because in 90% of the error cases of this error, this is likely the problem, is not it?

Reply:
Ah or this cannot be a config issue?